src/test/java/io/sniffy/socket/EchoServerRule.java

```diff
@@ -0,0 +1,190 @@
+                break;
+            } catch (IOException e) {
+                try {
+                    serverSocket.close();
+                } catch (Exception ex) {
+                    ex.printStackTrace();
+                }
+            }
+        }
+
+        if (null == serverSocket) {
+            throw new IOException("Failed to find an available port");
+        }
+
+        thread.start();
+
+    }
+
+    private CountDownLatch countDownLatch = new CountDownLatch(2);
+
+    public CountDownLatch getCountDownLatch() {
+        return countDownLatch;
+    }
+
+    @Override
+    public void run() {
+
+        try {
+            while (!Thread.interrupted()) {
+
+                Socket socket = serverSocket.accept();
+
+                sockets.add(socket);
+
+                InputStream inputStream = socket.getInputStream();
+                OutputStream outputStream = socket.getOutputStream();
+
+                Thread socketInputStreamReaderThread = new Thread(new SocketInputStreamReader(inputStream));
+                Thread socketOutputStreamWriterThread = new Thread(new SocketOutputStreamWriter(outputStream));
+
+                socketThreads.add(socketInputStreamReaderThread);
+                socketThreads.add(socketOutputStreamWriterThread);
+
+                socketInputStreamReaderThread.start();
+                socketOutputStreamWriterThread.start();
+            }
+        } catch (SocketException e) {
+            if (!"socket closed".equalsIgnoreCase(e.getMessage())) {
+                e.printStackTrace();
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    @Override
+    protected void after() {
+
+        socketThreads.forEach(Thread::interrupt);
+
+        joinThreads();
+
+    }
+
+    public void joinThreads() {
+
+        socketThreads.forEach((thread) -> {
+            try {
+                thread.join();
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        });
+
+        sockets.forEach((socket) -> {
+            try {
+                socket.close();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        });
+
+        thread.interrupt();
+
+        try {
+            serverSocket.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        try {
+            thread.join();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    private class SocketInputStreamReader implements Runnable {
+
+        private final InputStream inputStream;
+
+        public SocketInputStreamReader(InputStream inputStream) {
+            this.inputStream = inputStream;
+        }
+
+        @Override
+        public void run() {
+            try {
+
+                countDownLatch.countDown();
+
+                int totalRead = 0, read = 0;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Use one line for each declaration, it enhances code readability.](https://www.codacy.com/app/sniffy/sniffy/file/2251188807/issues/source?bid=3005960&fileBranchId=3034483#l147)